providers/base/bin/switch_power_mode.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# All rights reserved.
+#
+# Written by:
+#   Bin Li <bin.li@canonical.com>
+
+"""Modules providing a function running os or sys commands."""
+import os
+import sys
+
+
+def main():
+    """Switching the power mode to check if the power mode can be switched."""
+    sysfs_root = "/sys/firmware/acpi/"
+    if not os.path.isdir(sysfs_root):
+        return
+    choices_filename = os.path.join(sysfs_root, "platform_profile_choices")
+    if (not os.path.isfile(choices_filename) or
+            not os.access(choices_filename, os.R_OK)):
+        return
+    profile_filename = os.path.join(sysfs_root, "platform_profile")
+    if (not os.path.isfile(profile_filename) or
+            not os.access(profile_filename, os.R_OK)):
+        return
```

Comment:
1) Use `pathlib.Path` and not `os.path` here, it is way more handy
2) As the [documentation of `os.access` says](https://docs.python.org/3/library/os.html#os.access), don't check before accessing, try to open and catch `FileNotFoundError` and `PermissionError`
3) This exits the program with `$?` == 0 (so the test will pass), is this intended?

Reply:
@Hook25 
I  agree with you for the first two suggestions.
For the 3rd one, I don't wat the testcase report failure if the platform_profile_choices does not exist, and skip it, any other ideas for this one?

Reply:
What you need is a resource. A test in checkbox can either pass, fail or be skipped. To skip it you have to add a `requires` constraint to the pxu and a resource to require. 

The usual pattern to do this is to add to your script a command line arg that creates a resource output, something like:
```
switch_power_mode.py --resource
```

The code could be something like this:
```python
def print_resource():
    sysfs_root = Path("/sys/firmware/acpi/")
    choices_path = sysfs_root / "platform_profile_choices"
    profile_path = sysfs_root / "platform_profile"
    supported = (sysfs_root.exists()) and profile_path.exists() and choices_path.exists())
    print("acpi_profiles_supported: {}".format(supported)
```

Then you can create a resource job like the following
```yaml
id: switch_power_mode
plugin: resource
_summary:
    Discover if the system supports power modes via acpi
estimated_duration: 1s
command:
    switch_power_mode.py --resource
```

Finally in your pxu you can simply add a constraint like:
```yaml
requires:
    switch_power_mode.acpi_profiles_supported == 'True''
```

If the feature is not supported, your test will be skipped

Reply:
Done.